src/frontend/admin/src/components/templates/relations/course-product-relation/CourseProductRelationList.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from "react";
+import { useEffect, useMemo } from "react";
+import { defineMessages, FormattedMessage, useIntl } from "react-intl";
+import Stack from "@mui/material/Stack";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import { SimpleCard } from "@/components/presentational/card/SimpleCard";
+import { useModal } from "@/components/presentational/modal/useModal";
+import { CourseFormProductRelationModal } from "@/components/templates/courses/form/product-relation/CourseFormProductRelationModal";
+import { CourseProductRelationRow } from "@/components/templates/courses/form/sections/product-relation/CourseProductRelationRow";
+import { CourseProductRelation } from "@/services/api/models/Relations";
+import { CourseProductRelationDummyRow } from "@/components/templates/courses/form/sections/product-relation/CourseProductRelationDummyRow";
+import { AlertModal } from "@/components/presentational/modal/AlertModal";
+import { CustomList } from "@/components/presentational/list/CustomList";
+import { useCourseProductRelationList } from "@/components/templates/courses/form/sections/product-relation/useCourseProductRelationList";
+
+export enum CourseProductRelationSource {
+  PRODUCT = "product",
+  COURSE = "course",
+}
+
+const messages = defineMessages({
+  addRelationButtonLabel: {
+    id: "components.templates.relations.courseProductRelation.CourseProductRelationList.addRelationButtonLabel",
+    defaultMessage: "Add relation",
+    description: "Label for the product relation subheader form",
+  },
+  deleteRelationModalTitle: {
+    id: "components.templates.relations.courseProductRelation.CourseProductRelationList.deleteRelationModalTitle",
+    description: "Title for the delete relation modal",
+    defaultMessage: "Delete product relationship",
+  },
+  deleteRelationModalContent: {
+    id: "components.templates.relations.courseProductRelation.CourseProductRelationList.deleteRelationModalContent",
+    description: "Content for the delete relation modal",
+    defaultMessage: "Are you sure you want to delete this relation?",
+  },
+  emptyList: {
+    id: "components.templates.relations.courseProductRelation.CourseProductRelationList.emptyList",
+    description: "Message for empty list",
+    defaultMessage:
+      "{isCourse, select, true {No product relationships have been created for this course} other {No course relationships have been created for this product}}",
+  },
+  titleHeader: {
+    id: "components.templates.relations.courseProductRelation.CourseProductRelationList.titleHeader",
+    description: "Message for empty list",
+    defaultMessage:
+      "{isCourse, select, true {Relation to products} other {Relation to courses}}",
+  },
```

Comment:
IMO you are putting to much logic in translation message. Create two distinct messages will be simpler